### PR TITLE
feat: add kilo help --all command and auto-generated CLI reference docs

### LINKED ISSUE #6294: [FEATURE]: add `kilo help --all` to dump full CLI reference as markdown
### Is there an existing issue for this?

- [x] I have verified this feature I'm about to request hasn't been suggested before.

## Summary

Add a `kilo help` command that outputs the complete CLI reference — all commands and subcommands — as Markdown or plain text.

## Motivation

Useful for generating LLM context files, documentation, and quick offline reference. Inspired by Fossil SCM's `fossil help --all`.

## Proposed API

```
kilo help --all              # full reference, markdown (default)
kilo help --all --format text  # plain text
kilo help auth               # scoped to a single command + its subcommands
```

- Hidden/internal commands included with an `[internal]` callout
- Output is pipeable: `kilo help --all > REFERENCE.md`

### PR DESCRIPTION
**Pull request by maphew**
## Summary

- Add `kilo help --all` command that outputs full CLI reference in markdown or plain text format (#560)
- Add `generateCommandTable()` function that produces a markdown command table from the command barrel
- Auto-generate two docs artifacts via `script/generate.ts`: a Markdoc partial for the command table and a full CLI Command Reference page
- Replace the hand-written (stale) command table in `cli.md` with the generated partial
- Add CLI Command Reference nav entry in the docs site

Closes Kilo-Org/kilocode#6294, closes Kilo-Org/kilocode#6291

## Credit

The quick-reference table format was inspired by @Githubguy132010's contribution in Kilo-Org/kilocode#6294 — thanks for the idea!

### `kilo help` command
- **`src/kilocode/help.ts`** — Core `generateHelp()` + new `generateCommandTable()` functions
- **`src/kilocode/help-command.ts`** — `HelpCommand` yargs CommandModule
- **`src/cli/commands.ts`** — Barrel file exporting all CommandModule objects
- **`src/index.ts`** — Replaced 19 individual `.command()` calls with barrel import + loop

### Auto-generated docs
- **`script/generate-cli-docs.ts`** — Generation script (runs with `--conditions=browser`)
- **`src/kilocode/generate-cli-docs.ts`** — Entrypoint that writes both generated files
- **`script/generate.ts`** — Added CLI docs generation step before format
- **`packages/kilo-docs/markdoc/partials/cli-commands-table.md`** — Generated command table partial
- **`packages/kilo-docs/pages/code-with-ai/platforms/cli-reference.md`** — Generated full reference page

### Docs site updates
- **`packages/kilo-docs/pages/code-with-ai/platforms/cli.md`** — Replaced hand-written table with `{% partial %}`
- **`packages/kilo-docs/lib/nav/code-with-ai.ts`** — Added "Command Reference" subLink under CLI

## Usage

```bash
kilo help --all                    # Full CLI reference in Markdown
kilo help auth                     # Scoped to auth command + subcommands
kilo help --all --format text      # Plain text format
kilo help --all > REFERENCE.md     # Pipeable to file
```

## Testing

- 15 tests in `test/kilocode/help.test.ts` (no mocks) — markdown/text output, subcommands, filtering, ANSI stripping, table generation, edge cases
- Typecheck clean (`bun turbo typecheck`)
- Docs build succeeds with new `cli-reference` page

### CODE REVIEW

File: packages/opencode/src/kilocode/help.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+      const help = strip(await subInst.getHelp())
+      result.push({
+        name: full,
+        hidden: handler.description === false,
+        help,
+      })
+    }
+  } catch (err) {
+    // yargs internals unavailable
+  }
```
[the remark this answers is not in the document]
Comment (maphew) [reply packages/opencode/src/kilocode/help.ts:76]: Fixed in faecb4a — the catch block now logs via `console.warn`. Used `console.warn` rather than `Log` since this module runs before project initialization.

File: packages/opencode/src/kilocode/help.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+  const inst = yargs([]).scriptName(`kilo ${name}`).wrap(null)
+  ;(cmd.builder as any)(inst)
+
+  const result: Array<{ name: string; hidden: boolean; help: string }> = []
+
+  try {
+    const internal = (inst as any).getInternalMethods()
+    const cmdInstance = internal.getCommandInstance()
+    const handlers = cmdInstance.getCommandHandlers()
```
[the remark this answers is not in the document]
Comment (maphew) [reply packages/opencode/src/kilocode/help.ts:49]: Added a comment in faecb4a documenting this is verified against yargs@18.0.0 and noting the graceful degradation behavior (subcommand help omitted, top-level help still works). The catch block now also logs a warning so breakage won't be silent.

**Comment by maphew** [comment]
## Note on duplicate PR

There's also #570 targeting this issue. Both are valid approaches — here's a quick comparison for reviewers:

**#570** uses a static registry — a hand-written data structure (~250 lines) describing every command, option, and subcommand. The output is nicely formatted with Markdown tables for options and a version banner. The downside is that the registry must be manually updated whenever commands change, and nothing enforces that. A few descriptions have already drifted from the actual command definitions (e.g. `debug scrap` and `debug skill`).

**This PR (#571)** uses dynamic introspection — it walks the real `CommandModule` objects at runtime via yargs internals to generate help text. This means it automatically picks up any command added to or changed in the barrel, so it can't drift out of sync. The tradeoff is the output is less polished (code-fenced yargs help text rather than structured tables), and it relies on yargs internal APIs (documented in code, with graceful fallback if they change).

| | #570 (static) | #571 (dynamic) |
|---|---|---|
| Accuracy over time | Requires manual sync | Always matches reality |
| Output polish | Structured tables | Code-fenced help text |
| Maintenance cost | Update registry per command change | None |
| Fragility | None (static data) | Yargs internals (with fallback) |
| Commands barrel | Not extracted | Extracted for reuse |
| File location | `src/cli/cmd/` (upstream path) | `src/kilocode/` (lower merge conflict risk) |
| Testability | Monkey-patches stdout | Pure function, no patching |

Either approach works — the core question is whether the team prefers polished static output that needs maintenance, or auto-generated output that stays correct on its own.

File: packages/opencode/test/kilocode/help.test.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+})
+
+describe("edge cases", () => {
+  test("output contains no ANSI escape sequences", async () => {
+    const output = await generateHelp({ all: true, format: "md", commands })
+    expect(/\x1b\[/.test(output)).toBe(false)
+  })
+
+  test("kilo help nonexistent throws unknown command error", async () => {
+    expect(generateHelp({ command: "nonexistent", commands })).rejects.toThrow("unknown command")
```
[the remark this answers is not in the document]
Comment (maphew) [reply packages/opencode/test/kilocode/help.test.ts]: Fixed in 81b1904 — added `await` to the assertion.

File: packages/opencode/src/kilocode/help.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+export async function generateHelp(options: {
+  command?: string
+  all?: boolean
+  format?: "md" | "text"
+  commands?: Cmd[]
+}): Promise<string> {
+  const format = options.format ?? "md"
+
+  const all = options.commands ?? (await loadCommands())
```
[the remark this answers is not in the document]
Comment (maphew) [reply packages/opencode/src/kilocode/help.ts:165]: Fixed in 81b1904 — `options.all` now gates the filtering via an IIFE. When neither `all` nor `command` is set, `relevant` is empty and the function returns an empty string.

File: packages/kilo-docs/pages/code-with-ai/platforms/cli-reference.md [hunk 5]
```diff
@@ -0,0 +1,557 @@
...
+
+Options:
+  --help         Show help  [boolean]
+  --version      Show version number  [boolean]
+  --port         port to listen on  [number] [default: 0]
+  --hostname     hostname to listen on  [string] [default: "127.0.0.1"]
+  --mdns         enable mDNS service discovery (defaults hostname to 0.0.0.0)  [boolean] [default: false]
+  --mdns-domain  custom domain name for mDNS service (default: opencode.local)  [string] [default: "opencode.local"]
+  --cors         additional domains to allow for CORS  [array] [default: []]
+  --cwd          working directory  [string] [default: "/var/home/matt/dev/kilo/packages/opencode"]
```
[the remark this answers is not in the document]
Comment (maphew) [reply packages/kilo-docs/pages/code-with-ai/platforms/cli-reference.md]: Fixed in 81b1904 — the generation script now sanitizes `process.cwd()` paths with `.replaceAll(cwd, ".")` before writing the output. The `--cwd` default now shows `[default: "."]` instead of the developer's local path.

File: packages/opencode/test/kilocode/help.test.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  ServeCommand,
+  WebCommand,
+  ModelsCommand,
+  StatsCommand,
+  ExportCommand,
+  ImportCommand,
+  PrCommand,
+  SessionCommand,
+  HelpCommand,
+] as any[]
```
[the remark this answers is not in the document]
Comment (maphew) [reply packages/opencode/test/kilocode/help.test.ts:58]: Fixed in 81b1904 — added `AttachStub` (same pattern as `TuiStub`) to the test commands array. Can't import the real `AttachCommand` because it's under `tui/` which pulls in `@opentui/solid` and fails in the test environment.

**Comment by markijbema** [comment]
Hi! Thank you for taking the time to contribute to this project—we really appreciate it. 🙏 

We are currently working on re-platforming the core of our VS Code and JetBrains extensions to be based on our new Kilo CLI, with a complete rebuild based on OpenCode as our new foundation, and the moment has come to promote this repository to become the main repository. To do that, we moved the code from this repository to the kilocode repository.

This unfortunately means we cannot merge this branch here anymore. Please add https://github.com/Kilo-Org/kilocode.git as a remote, and push your branch there and create a new PR in https://github.com/Kilo-Org/kilocode . We unfortunately cannot do this for you as then the PR would not be in your name anymore. If you need any help, feel free to ask on our Discord in #kilo-dev-contributors

Sorry for the inconvenience and thank you for contributing to Kilo!
